RealmSwift/RealmCollection.swift

```diff
@@ -1236,6 +1235,17 @@ public struct AnyRealmCollection<Element: RealmCollectionValue>: RealmCollection
     fileprivate init(base: _AnyRealmCollectionBase<Element>) {
         self.base = base
     }
+    fileprivate init(rlmCollection: RLMCollection) {
+        if let rlmCollection = rlmCollection as? RLMArray<AnyObject> {
+            self.base = _AnyRealmCollection(base: List<Element>(objc: rlmCollection))
+        } else if let rlmCollection = rlmCollection as? RLMSet<AnyObject> {
+            self.base = _AnyRealmCollection(base: MutableSet<Element>(objc: rlmCollection))
+        } else if let rlmCollection = rlmCollection as? RLMResults<AnyObject> {
+            self.base = _AnyRealmCollection(base: Results<Element>(objc: rlmCollection))
+        } else {
+            preconditionFailure()
+        }
+    }
```

Comment:
Linking objects also extends RealmCollection. Does it need to be here?

Reply:
`LinkingObjects` is bridged to `RLMCollection` as an `RLMResults`.